### PR TITLE
Fix syntax highlight issues for arrow functions and function parameters

### PR DESCRIPTION
**Pull request by zhengbli**
Fix the issue that arrow function is not highlighted correctly (credit to @billti )
a

Also moved the equal sign ``=`` to the correct scope.

### CODE REVIEW

File: TypeScript.YAML-tmLanguage [hunk 1]
```diff
@@ -402,11 +405,11 @@ repository:
 
   relational-operator:
     name: keyword.operator.comparison.ts
-    match: ===|==|=|!=|!==|<=|>=|<>|<|>
+    match: ===|==|!=|!==|<=|>=|<>|<|>
```
Comment (billti) [inline TypeScript.YAML-tmLanguage]: "!==" should go before "!=", otherwise "!=" will always match first.

File: TypeScript.YAML-tmLanguage [hunk 2]
```diff
@@ -273,8 +273,10 @@ repository:
     end: '\)'
     patterns:
     - include: '#comment'
+    - include: '#parameter-name'
```
Comment (billti) [inline TypeScript.YAML-tmLanguage]: This looks like a non-trivial change that could have quite an impact.  Let's chat about this in person tomorrow before you merge anything.

Comment (zhengbli) [inline TypeScript.YAML-tmLanguage]: Yeah sure. I have also written a test file the listed all the common cases for scope detection which I can show you tomorrow.

File: TypeScript.YAML-tmLanguage (hunk 1 above)
Comment (billti) [inline TypeScript.YAML-tmLanguage]: Bump on this... did this ever get corrected? (Or a test added and me proved wrong? :smile: )

**Pull request closed without merging**